lib/core/text/abstract_text.nit

```diff
@@ -2061,6 +2061,37 @@ redef class Float
 	# assert 0.999.to_precision(4)     == "0.9990"
 	# ~~~
 	fun to_precision(decimals: Int): String
+	do
+		var check = check_nan_inf
+		if check != null then return check
+
+		var size = to_precision_size(decimals)
+		var cstr = new CString(size+1)
+		to_precision_fill(decimals, size+1, cstr)
+		return cstr.to_s_unsafe(byte_length=size, copy=false)
+	end
+
+	# Return Hexadecimal `String` representation of `self` in exponential notation
+	#
+	# ~~~
+	# assert 12.345.to_hexa_exponential_notation    == "0x1.8b0a3d70a3d71p+3"
+	# assert 12.345.to_hexa_exponential_notation.to_f == 12.345
+	# ~~~
+	fun to_hexa_exponential_notation: String
+	do
+		var check = check_nan_inf
+		if check != null then return check
+
+		var size = to_precision_size_hexa
+		var cstr = new CString(size+1)
+		to_precision_fill_hexa(size+1, cstr)
+		return cstr.to_s_unsafe(byte_length=size, copy=false)
+	end
+
+	# Check if `self` is an nan or inf.
```

Comment:
```suggestion
	# Is `self` is not a number or infinity?
```

Reply:
Shouldn't it be the behaviour of `to_s`? Not sure why's this method needed?

Reply:
I completely forgot, i deal with hexadecimal representation. It is not needed to return `not a number` or` infinity`. This factorisation is not necessary i remove it an place it in the original method (to_precision). For the to_s, it doesn't do this check because it calls to_precision and this method do this check for it.